File: 2016/Midterm/p4.c
/*
 * In The Name Of God
 * ========================================
 * [] File Name : p4.c
 *
 * [] Creation Date : 01-12-2016
 *
 * [] Created By : Parham Alvani (parham.alvani@gmail.com)
 * =======================================
*/
/*
 * Copyright (c) 2016 Parham Alvani.
*/
#include <stdio.h>

int fibonacci(int n)
{
	if (n == 1 || n == 2)
		return 1;
	else
		return fibonacci(n - 1) + fibonacci(n - 2);
}

int fibonacci_check_and_generate(int n)
{
	int i = 1;

	while (fibonacci(i) < n)
		i++;

	if (fibonacci(i) == n)
		return fibonacci(i + 1);
	else
		return -1;
}

int main(int argc, char *argv[])
{
	int n;
	int retval;

	scanf("%d", &n);

	retval = fibonacci_check_and_generate(n);
	if (retval == -1)
		printf("Error !\n");
	else
		printf("%d %d\n", retval - n, retval);
}

File: 2017/HW-3/p4.go
/*
 * +===============================================
 * | Author:        Parham Alvani <parham.alvani@gmail.com>
 * |
 * | Creation Date: 03-11-2017
 * |
 * | File Name:     p4.go
 * +===============================================
 */

package main

import "fmt"

func main() {
	var n int

	fmt.Scanf("%d", &n)

	if n < 10000 || n > 99999 {
		fmt.Printf("Error\n")
		return
	}

	a := n % 10
	n /= 10
	b := n % 10
	n /= 10
	n /= 10
	d := n % 10
	n /= 10
	e := n % 10

	if a == e && b == d {
		fmt.Printf("True\n")
	} else {
		fmt.Printf("False\n")
	}

}
